specification/sql/resource-manager/Microsoft.Sql/preview/2023-05-01-preview/LocationCapabilities.json

```diff
@@ -579,6 +580,119 @@
+          "readOnly": true
+        },
+        "supportedServiceLevelObjectives": {
+          "description": "The list of supported service level objectives for the edition.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobAgentServiceLevelObjectiveCapability"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": []
+        },
+        "status": {
+          "description": "The status of the capability.",
+          "enum": [
+            "Visible",
+            "Available",
+            "Default",
+            "Disabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "CapabilityStatus",
+            "modelAsString": false
+          }
+        },
+        "reason": {
+          "description": "The reason for the capability not being available.",
+          "type": "string"
+        }
+      }
+    },
+    "JobAgentServiceLevelObjectiveCapability": {
+      "description": "The job agent service level objective capability.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The service objective name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "sku": {
+          "$ref": "../../../common/v1/types.json#/definitions/Sku",
+          "description": "The sku.",
+          "readOnly": true
+        },
+        "status": {
+          "description": "The status of the capability.",
+          "enum": [
+            "Visible",
+            "Available",
+            "Default",
+            "Disabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "CapabilityStatus",
+            "modelAsString": false
+          }
+        },
+        "reason": {
+          "description": "The reason for the capability not being available.",
+          "type": "string"
+        }
+      }
+    },
+    "JobAgentVersionCapability": {
+      "description": "The job agent version capability.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The job agent version name.",
+          "type": "string",
+          "readOnly": true
+        },
+        "supportedEditions": {
+          "description": "The list of supported editions.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/JobAgentEditionCapability"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": []
+        },
+        "status": {
+          "description": "The status of the capability.",
+          "enum": [
+            "Visible",
+            "Available",
+            "Default",
+            "Disabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "CapabilityStatus",
+            "modelAsString": false
+          }
+        },
+        "reason": {
```

Comment:
>reason [](http://example.com/codeflow?start=9&length=6)

This appears to only apply for certain status values. What is it set to for other status values? Empty string? Something else? Consider making this more general, e.g. "details" or "detailedStatus", so it makes sense for any/all status values. Same comment applies for 2 other "reason" properties.

Reply:
As with the name property we are trying to be consistent with all of the other location capabilities that have been defined this way for several API versions.